### PR TITLE
Making the gRPC C tests building under Windows.

### PR DESCRIPTION
**Pull request by nicolasnoble**

### CODE REVIEW

**Comment by ctiller** [comment]
Can you add a comment to grpc.mak.template explaining the filter for which
tests go in and which get excluded?

**Comment by nicolasnoble** [comment]
You mean the do_not_build array ?

File: templates/vsprojects/Grpc.mak.template [hunk 1]
```diff
@@ -33,10 +33,19 @@
...
-                        if not disallowed_dependencies.intersection(target.deps) and
+  do_not_build = set(['gpr',
+                      'grpc',
+                      'grpc_unsecure',
+                      'gpr_test_util',
+                      'grpc_test_util',
+                      'grpc_test_util_unsecure',
+                     ])
+  buildable_targets = [ target for target in targets + libs
+                        if not disallowed_dependencies.intersection(target.get('deps', [])) and
```
Comment (ctiller) [inline templates/vsprojects/Grpc.mak.template]: This if clause needs a comment

File: templates/vsprojects/Grpc.mak.template [hunk 2]
```diff
@@ -33,10 +33,19 @@
 <%def name="to_windows_path(path)">${path.replace('/','\\')}</%def>\
 <%
   disallowed_dependencies = set(['end2end_certs'])
-  buildable_targets = [ target for target in targets
-                        if not disallowed_dependencies.intersection(target.deps) and
+  do_not_build = set(['gpr',
```
Comment (ctiller) [inline templates/vsprojects/Grpc.mak.template]: This list needs a brief explanation as to why

**Comment by ctiller** [comment]
Added line comments now that I have hands free for a laptop :)

**Comment by nicolasnoble** [comment]
How about that ? :-)

**Comment by nicolasnoble** [comment]
Bunch of changes in that PR.

**Pull request merged**